[ios][android] Add wheelchair info to place page

dvdmrtnz opened the pull request:
This PR adds wheelchair info to the place page allowing to distinguish between `wheelchair=no`, `wheelchair=limited` and `wheelchair=yes`.

https://wiki.openstreetmap.org/wiki/Key:wheelchair

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/f868a30e-2d0f-4162-8ee2-7b334955a34e" width="320"/>

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/4a2acfae-ba11-434b-9fa5-7b512f3ae47a" width="320"/>

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/92e19f3b-70dc-40c2-800a-978443945602" width="320"/>

vng reviewed (commented):
LGTM. Android will be later or you don't want to dive into it?

dvdmrtnz commented:
@vng I have not added it to Android because I am not sure how to access the `GetLocalizedWheelchairType()` function from Android. If you have some suggestions on how to handle it I can give it a go. Otherwise this can be added just to iOS at the moment.

dvdmrtnz commented:
@vng I feel like a refactoring here could help with adding these attributes that are derived from the classificator types. Maybe they could be generated in core c++ code and then passed as an array of tags, a bit like you said in PR https://github.com/organicmaps/organicmaps/pull/7108#issuecomment-1890429469.

vng commented:
Well, definitely we can implement an array with a predefined params structure to add similar-classifier attributes here.
You can try or I can try a bit later. At this moment your solution is good and acceptable.

biodranik commented on indexer/feature_utils.cpp:
```diff
@@ -438,6 +438,16 @@ string GetLocalizedFeeType(TypesHolder const & types)
+string GetLocalizedWheelchairType(TypesHolder const & types)
+{
+  auto const & isWheelchairType = ftypes::IsWheelchairTypeChecker::Instance();
+  auto localized_types = GetLocalizedTypes(isWheelchairType, types);
```
```suggestion
  auto localizedTypes = GetLocalizedTypes(isWheelchairType, types);
```

biodranik replied:
Underscores are used only in member prefixed variables starting from m_

biodranik commented on iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm:
```diff
@@ -83,6 +83,9 @@ - (instancetype)initWithRawData:(Info const &)rawData ohLocalization:(id<IOpenin
     _atm = rawData.HasAtm() ? NSLocalizedString(@"type.amenity.atm", nil) : nil;
       
+    std::string wheelchair = rawData.GetLocalizedWheelchairType();
+    _wheelchair = !wheelchair.empty() ? ToNSString(wheelchair) : nil;
```
Isn't _wheelchair always nil by default?
```suggestion
    if (!wheelchair.empty())
      _wheelchair = @(wheelchair.c_str());
```

biodranik commented on iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm:
```diff
@@ -83,6 +83,9 @@ - (instancetype)initWithRawData:(Info const &)rawData ohLocalization:(id<IOpenin
     
     _atm = rawData.HasAtm() ? NSLocalizedString(@"type.amenity.atm", nil) : nil;
       
+    std::string wheelchair = rawData.GetLocalizedWheelchairType();
```
How editing of wheelchair accessibility will be supported? Usually, when we add some fields into Place Page, we assume that they are/will be editable.

Does it make sense to introduce and pass an enum from the C++ core, and localize it in the UI? Or just some hard-coded keys-strings "unknown", "yes", "no", and "limited"?

Also think about Android implementation to make it easier. Maybe something similar to already existing editor implementations like WiFi editing?

dvdmrtnz commented:
I have changed how this works. I have added a new metadata property that gets filled at runtime. This way I can add this to android the same way as the rest of the properties.

Also the metadata uses the readable wheelchair type and gets localized in the UI.

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/f1f4b368-562a-4a4e-9ef7-2520c37327f9" width="320"/>

vng commented on android/app/src/main/java/app/organicmaps/bookmarks/data/Metadata.java:
```diff
@@ -60,7 +60,8 @@ public enum MetadataType
     FMD_WIKIMEDIA_COMMONS(41),
-    FMD_CAPACITY(42);
+    FMD_CAPACITY(42),
+    FMD_WHEELCHAIR(43);
```
This approach is ok I think. Just to add a comment here: The value is runtime only, but actual data is taken from the classifier types.

I suppose we will remove it from the classifier someday and add it to the metadata.

vng commented:
@rtsisyk Why do we have _legal_ group in reviewers here? :)

Jean-BaptisteC commented:
> @rtsisyk Why do we have _legal_ group in reviewers here? :)

Because this PR update files in assets directory

dvdmrtnz commented on indexer/feature_meta.hpp:
```diff
@@ -148,6 +148,7 @@ class Metadata : public MetadataBase
     FMD_BUILDING_MIN_LEVEL = 40,
     FMD_WIKIMEDIA_COMMONS = 41,
     FMD_CAPACITY = 42,
+    FMD_WHEELCHAIR = 43,
```
I added a comment
